crates/editor/src/editor.rs

```diff
@@ -3980,6 +3996,71 @@ impl Editor {
         }))
     }
 
+    fn hide_signature_help(&mut self, cx: &mut ViewContext<Self>) -> bool {
+        if self.signature_help_state.is_some() {
+            self.signature_help_state = None;
+            self.signature_help_task = None;
+            cx.notify();
+            true
+        } else {
+            false
+        }
+    }
+
+    pub fn show_signature_help(&mut self, _: &ShowSignatureHelp, cx: &mut ViewContext<Self>) {
+        if self.pending_rename.is_some() {
+            return;
+        }
+
+        let position = self.selections.newest_anchor().head();
+        let Some((buffer, buffer_position)) =
+            self.buffer.read(cx).text_anchor_for_position(position, cx)
+        else {
+            return;
+        };
+
+        self.signature_help_task = Some(cx.spawn(move |editor, mut cx| async move {
+            let markdown_task_result = editor.update(&mut cx, |editor, cx| {
+                let maybe_language = editor.language_at(0, cx);
```

Comment:
```suggestion
                let maybe_language = editor.language_at(position, cx);
```

`0` is the beginning of the document, which might be a totally different document, imagine a Markdown document with TS code snippets.

Reply:
@SomeoneToIgnore 
Thank you! Fixed it on c68746c8d